File: software/examples/php/ExampleWriteReadType2.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletNFC.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletNFC;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your NFC Bricklet

// Callback function for state changed callback
function cb_stateChanged($state, $idle, $user_data)
{
	$nfc = $user_data;

	if($state == BrickletNFC::READER_STATE_IDLE) {
		$nfc->readerRequestTagID();
	}
	else if ($state == BrickletNFC::READER_STATE_REQUEST_TAG_ID_READY) {
		$tag_id = [];
		$ret = $nfc->readerGetTagID();

		if ($ret["tag_type"] != BrickletNFC::TAG_TYPE_TYPE2) {
			echo "Tag is not type-2\n";

			return;
		}

    printf("Found tag of type %u with ID [0x%X 0x%X 0x%X 0x%X]\n",
		       $ret["tag_type"],
					 $ret["tag_id"][0],
					 $ret["tag_id"][1],
					 $ret["tag_id"][2],
					 $ret["tag_id"][3]);

		$nfc->readerRequestPage(1, 4);
	}
	else if ($state == BrickletNFC::READER_STATE_REQUEST_TAG_ID_ERROR) {
		echo "Request tag ID error\n";
	}
	else if ($state == BrickletNFC::READER_STATE_REQUEST_PAGE_READY) {
		$page = $nfc->readerReadPage();
		printf("Page read: 0x%X 0x%X 0x%X 0x%X\n", $page[0], $page[1], $page[2], $page[3]);
    $nfc->readerWritePage(1, $page);
	}
	else if ($state == BrickletNFC::READER_STATE_WRITE_PAGE_READY) {
		echo "Page write OK\n";
	}
	else if ($state == BrickletNFC::READER_STATE_REQUEST_PAGE_ERROR) {
		echo "Request page error\n";
	}
	else if ($state == BrickletNFC::READER_STATE_WRITE_PAGE_ERROR) {
		echo "Request write page error\n";
	}
}

$ipcon = new IPConnection(); // Create IP connection
$nfc = new BrickletNFC(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register state changed callback to function cb_stateChanged
$nfc->registerCallback(BrickletNFC::CALLBACK_READER_STATE_CHANGED, 'cb_stateChanged', $nfc);

$nfc->setMode(BrickletNFC::MODE_READER);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/go/example_write_read_type2.go
package main

import (
	"fmt"
    "github.com/tinkerforge/go-api-bindings/ipconnection"
    "github.com/tinkerforge/go-api-bindings/nfc_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your NFC Bricklet.

func main() {
	ipcon := ipconnection.New()
    defer ipcon.Close()
	nfc, _ := nfc_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
    defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	nfc.RegisterReaderStateChangedCallback(func(state nfc_bricklet.ReaderState, idle bool) {
		switch state {
		case nfc_bricklet.ReaderStateIdle:
			nfc.ReaderRequestTagID()
		case nfc_bricklet.ReaderStateRequestTagIDReady:
			tagID, tagType, _ := nfc.ReaderGetTagID()
			if tagType != nfc_bricklet.TagTypeType2 {
				fmt.Println("Tag is not type-2")
				return				
			}

			fmt.Printf("Found tag of type %d with ID %v", tagType, tagID)
		case nfc_bricklet.ReaderStateRequestPageReady:
			page, _ := nfc.ReaderReadPage()
			fmt.Printf("Page read: % x", page)
			nfc.ReaderWritePage(1, page)
		case nfc_bricklet.ReaderStateWritePageReady:
			fmt.Println("Write page ready");
		case nfc_bricklet.ReaderStateRequestPageError:
			fmt.Println("Request page error");
		case nfc_bricklet.ReaderStateWritePageError:
			fmt.Println("Write page error");
		}
	})

	// Enable reader mode
	nfc.SetMode(nfc_bricklet.ModeReader)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

	ipcon.Disconnect()
}
